SpatialGDK/Source/SpatialGDKTests/SpatialGDK/RPCService/RPCServiceTest.cpp

```diff
@@ -0,0 +1,519 @@
+
+			if (Update.Update.component_id != ExpectedUpdateComponentId ||
+				!CompareUpdateToSendAndEntityPayload(Update, EntityPayloadItem))
+			{
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	bool ReceiveRPCTest(const TArray<Worker_EntityId>& EntityIdArray, ERPCEndpointType RPCEndpointType, ERPCType RPCType, const EntityPayload& EntityPayloadItem)
+	{
+		USpatialStaticComponentView* StaticComponentView = NewObject<USpatialStaticComponentView>();
+		for (Worker_EntityId EntityId : EntityIdArray)
+		{
+			Worker_Authority ClientAuth = GetClientAuthorityFromRPCEndpointType(RPCEndpointType);
+			Worker_Authority ServerAuth = GetServerAuthorityFromRPCEndpointType(RPCEndpointType);
+
+			Schema_ComponentData* ClientComponentData = Schema_CreateComponentData();
+			Schema_ComponentData* ServerComponentData = Schema_CreateComponentData();
+			Schema_Object* ClientSchemaObject = Schema_GetComponentDataFields(ClientComponentData);
+			Schema_Object* ServerSchemaObject = Schema_GetComponentDataFields(ServerComponentData);
+
+			if (EntityPayloadItem.EntityId == EntityId)
+			{
+				SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, RPCType, 1, EntityPayloadItem.Payload);
+				SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ServerSchemaObject, RPCType, 1, EntityPayloadItem.Payload);
+			}
+
+			AddEntityComponentToStaticComponentView(*StaticComponentView,
+				SpatialGDK::EntityComponentId(EntityId, SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID),
+				ClientComponentData,
+				ClientAuth);
+
+			AddEntityComponentToStaticComponentView(*StaticComponentView,
+				SpatialGDK::EntityComponentId(EntityId, SpatialConstants::SERVER_ENDPOINT_COMPONENT_ID),
+				ServerComponentData,
+				ServerAuth);
+		}
+
+		bool bExtractCallbackSuccesful = true;
+		ExtractRPCDelegate RPCDelegate = ExtractRPCDelegate::CreateLambda([&bExtractCallbackSuccesful, EntityPayloadItem](Worker_EntityId EntityId, ERPCType RPCType, const SpatialGDK::RPCPayload& Payload) {
+			bExtractCallbackSuccesful &= CompareRPCPayload(Payload, EntityPayloadItem.Payload);
+			bExtractCallbackSuccesful &= EntityId == EntityPayloadItem.EntityId;
+			return true;
+		});
+
+		TUniquePtr<SpatialGDK::SpatialRPCService> RPCService = CreateRPCService(EntityIdArray, RPCEndpointType, RPCDelegate, StaticComponentView);
+
+		Worker_ComponentId ComponentId = SpatialConstants::SERVER_ENDPOINT_COMPONENT_ID;
+		switch (RPCType)
+		{
+		case ERPCType::ClientReliable:
+		case ERPCType::ClientUnreliable:
+			ComponentId = SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID;
+			break;
+		case ERPCType::NetMulticast:
+			ComponentId = SpatialConstants::MULTICAST_RPCS_COMPONENT_ID;
+			break;
+		default:
+			break;
+		}
+
+		RPCService->ExtractRPCsForEntity(EntityPayloadItem.EntityId, ComponentId);
+		return bExtractCallbackSuccesful;
+	}
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_server_endpoint_WHEN_push_client_reliable_rpcs_on_the_service_THEN_check_success)
```

Comment:
Check success doesn't really mean anything. Maybe returns true?

Reply:
"success" is the SpatialGDK::EPushRPCResult we expect from the RPCService::PushRPC. It does mean something but maybe still needs to change. Perhaps it should be push_result_success.